app/assets/javascripts/app/views/notifications_view.js

```diff
@@ -70,7 +70,8 @@ app.views.Notifications = Backbone.View.extend({
 
     [allNotes, typeNotes, headerBadge].forEach(function(element){
       element.text(function(i, text){
-        return parseInt(text) + change });
+        return Math.max(parseInt(text) + change, 0);
```

Comment:
I got some negative results after playing with the notification dropdown for some time. As @AugierLe42e wrote there are still some improvements needed but this one was an easy fix to improve the UX. Otherwise you could mark some notifications as unread and the counter would still be negative or zero so you won't see it.